src/components/core/geoChart/legends/PushpinCategoryLegend.tsx

```diff
@@ -1,26 +1,98 @@
 // (C) 2020 GoodData Corporation
 import * as React from "react";
+import { ContentRect } from "react-measure";
+
+import FluidLegend from "../../../visualizations/chart/legend/FluidLegend";
 import StaticLegend from "../../../visualizations/chart/legend/StaticLegend";
+import { LEFT, RIGHT } from "../../../visualizations/chart/legend/PositionTypes";
+import { ChartType } from "../../../../constants/visualizationTypes";
 import { IPushpinCategoryLegendItem } from "../../../../interfaces/GeoChart";
 
+export const HEIGHT_OF_SIZE_LEGEND = 161;
+const PUSHPIN_TYPE: ChartType = "pushpin";
+
 export interface IPushpinCategoryLegendProps {
-    categoryItems: IPushpinCategoryLegendItem[];
-    position: string;
-    onItemClick(item: IPushpinCategoryLegendItem): void;
+    categoryItems?: IPushpinCategoryLegendItem[];
+    chartType?: ChartType;
+    contentRect: ContentRect;
+    format?: string;
+    hasSizeLegend: boolean;
+    height?: number;
+    locale?: string;
+    position?: string;
+    responsive?: boolean;
+    showFluidLegend?: boolean;
+    onItemClick?(item: IPushpinCategoryLegendItem): void;
 }
 
 export default function PushpinCategoryLegend(props: IPushpinCategoryLegendProps): JSX.Element {
-    const { position, categoryItems, onItemClick } = props;
-    const classNames = `viz-static-legend-wrap s-geo-category-legend position-${position}`;
-    const DEFAULT_HEIGHT = 0;
+    const { contentRect, hasSizeLegend, responsive, showFluidLegend } = props;
+    const isFluidLegendEnabled = responsive && showFluidLegend;
```

Comment:
Consider move to Geo util, I saw we reuse a lot
https://github.com/gooddata/gooddata-react-components/pull/1466/files#diff-401c51ce84aba39a6e5b2ba4c990ca86R200
https://github.com/gooddata/gooddata-react-components/pull/1466/files#diff-de6ab458c862f711882696296630b813R74

Reply:
updated